### ISSUE TITLE
Error with to_msgpack method trying to upgrade to 3.1 

### ISSUE THREAD

**Issue by gshank** [Fatal1ty/mashumaro#76]
* mashumaro version: 3.1
* Python version:  3.8.6
* Operating System: Mac OS

### Description

We're trying to convert from mashumaro 2.9 to 3.1., but our to_msgpack calls are failing with:
```
/Users/gerda/FTA/dbt/core/dbt/parser/manifest.py:549: in write_manifest_for_partial_parse
    manifest_msgpack = self.manifest.to_msgpack()
/Users/gerda/FTA/dbt/env/lib/python3.8/site-packages/mashumaro/mixins/msgpack.py:44: in to_msgpack
    self.to_dict(**dict(DEFAULT_DICT_PARAMS, **to_dict_kwargs))
<string>:139: in to_dict
    ???
<string>:21: in to_dict
    ???
_ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _

.0 = <dict_itemiterator object at 0x110a00630>

>   ???
E   TypeError: to_dict() got an unexpected keyword argument 'dialect'

<string>:21: TypeError
-------------------------------------------------------------------------------------------------- Captured stdout setup --------------------------------------------------------------------------------------------------


```

It looks like one of the generated to_dict() methods doesn't support the 'dialect' keyword.

What is the best way to debug this?

You can look at branch 'ct-413-bump_mashumaro' in the dbt core repo https://github.com/dbt-labs/dbt-core, and execute `pytest -v tests/functional/basic/test_basic.py` to see the error.

**Comment by Fatal1ty** [comment]
Hi @gshank

You're right, this exception says that dialect support is not enabled. You need to be sure that dialect support is enabled for the class that `to_dict` method is used with. I didn't find any use of `ADD_DIALECT_SUPPORT` in your code base.

See https://github.com/Fatal1ty/mashumaro#dialects

**Comment by Fatal1ty** [comment]
@gshank Hi, did you solve this problem on your side?

**Comment by Fatal1ty** [comment]
Fixed in 3.0.3.
